pwndbg/commands/vmmap.py

```diff
@@ -17,27 +20,38 @@
 import pwndbg.vmmap
 
 
-@pwndbg.commands.QuietSloppyParsedCommand
+def map_type(s):
+    gdbval_or_str = pwndbg.commands.sloppy_gdb_parse(s)
+
+    # returns a module filter
+    if isinstance(gdbval_or_str, six.string_types):
+        module_name = gdbval_or_str
+        return lambda page: module_name in page.objfile
+
+    # returns an address filter
+    elif isinstance(gdbval_or_str, six.integer_types + (gdb.Value,)):
+        addr = gdbval_or_str
+        return lambda page: addr in page
+
+    else:
+        raise argparse.ArgumentTypeError('Unknown vmmap argument type.')
+
+parser = argparse.ArgumentParser()
+parser.description = 'Print the virtual memory map, or the specific mapping for the provided address / module name.'
+parser.add_argument('map', type=map_type, nargs='?', default=None,
+                    help='Address or module name.')
+
+
+@pwndbg.commands.ArgparsedCommand(parser)
 @pwndbg.commands.OnlyWhenRunning
 def vmmap(map=None):
-    """
-    Print the virtal memory map, or the specific mapping for the
-    provided address / module name.
-    """
-    int_map = None
-    str_map = None
-
-    if isinstance(map, six.string_types):
-        str_map = map
-    elif isinstance(map, six.integer_types + (gdb.Value,)):
-        int_map = int(map)
+    print(map)
+    pages = list(filter(map, pwndbg.vmmap.get()))
```

Comment:
Never mind, I see now that `map` is actually a `lambda`. 

Can we change the name of the variable? It's overloading a keyword already, and is genuinely misleading in any case since it is not a `Page` (virtual memory map) nor a `{}` object.